### ISSUE TITLE
gdc_loadfile does not observe or respect datestamps properly

### ISSUE THREAD

**Issue by noblem** [broadinstitute/gdctools#57]
Today (2017_09_11) I tried ``make test_loadfile`` from a directory tree in which I had not done `test_mirror` and `test_dice` since 2017_07_19:

ls sandbox/mirror/TCGA/TCGA-ACC/metadata/
2017_07_19

But, instead of gdc_loadfile discovering the 2017_07_09 data and using it, it complained that there was no 2017_09_11 data instead!

```2017-09-11 18:33:46,243[INFO]: Inspecting data for TCGA-ACC version 2017_09_11
2017-09-11 18:33:46,245[ERROR]: Create Loadfile FAILED:
Traceback (most recent call last):
  File "../gdctools/gdc_loadfile.py", line 321, in execute
    (projects, annotations) = self.inspect_data()
  File "../gdctools/gdc_loadfile.py", line 113, in inspect_data
    metapath = get_diced_metadata(projname, projpath, self.datestamp)
  File "../gdctools/gdc_loadfile.py", line 357, in get_diced_metadata
    raise ValueError("Could not find dice metadata for " + project + " on " + datestamp)
ValueError: Could not find dice metadata for TCGA-ACC on 2017_09_11```

**Comment by noblem** [comment]
This is related to #47

**Comment by noblem** [comment]
Fixed in v0.2.5, commit https://github.com/broadinstitute/gdctools/commit/df437aba5baf8d4204a45577d4968ba6bf4e15fe
